From 02b222a335527f1031cc9495d8c5ebc1bc5b1d4e Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Wed, 11 Nov 2020 15:00:47 +0100
Subject: [PATCH] quota: Use realloc(3) instead of reallocarray(3)

reallocarray(3) has been added to glibc relatively recently (version
2.26, from 2017) and apparently not all users run new enough glibc. Just
use realloc(3) for now since in this case there's no real risk of
overflow.

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Signed-off-by: Jan Kara <jack@suse.cz>
Upstream-Status: Backport
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 quota.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/quota.c b/quota.c
index a6ed61f..a60de12 100644
--- a/quota.c
+++ b/quota.c
@@ -385,7 +385,7 @@ int main(int argc, char **argv)
 			  break;
 		  case 259:
 			  fscount++;
-			  fsnames = reallocarray(fsnames, fscount, sizeof(char *));
+			  fsnames = realloc(fsnames, fscount * sizeof(char *));
 			  if (!fsnames)
 				die(1, _("Not enough memory for filesystem names"));
 			  fsnames[fscount - 1] = optarg;
-- 
2.17.1

